api/envoy/api/v2/cluster/outlier_detection.proto

```diff
@@ -80,4 +80,20 @@ message OutlierDetection {
   // used to disable ejection or to ramp it up slowly. Defaults to 0.
   google.protobuf.UInt32Value enforcing_consecutive_gateway_failure = 11
       [(validate.rules).uint32.lte = 100];
+
+  // The number of consecutive locally originated failures before ejection
+  // occurs. Defaults to 5.
+  google.protobuf.UInt32Value consecutive_local_origin_failure = 12;
+
+  // The % chance that a host will be actually ejected when an outlier status
+  // is detected through consecutive locally originated failures. This setting can be
+  // used to disable ejection or to ramp it up slowly. Defaults to 100.
+  google.protobuf.UInt32Value enforcing_consecutive_local_origin_failure = 13
+      [(validate.rules).uint32.lte = 100];
+
+  // The % chance that a host will be actually ejected when an outlier status
+  // is detected through success rate statistics for locally originated errors.
+  // This setting can be used to disable ejection or to ramp it up slowly. Defaults to 100.
+  google.protobuf.UInt32Value enforcing_local_origin_success_rate = 14
```

Comment:
I'm a little confused here. There is now an enforcing setting for local origin success rate, but no setting for local origin success rate? Where is the local origin success rate configured?

At a high level, I'm confused as to whether this PR is going to have user visible effect for people that make no config changes. Will it? Either way, can you add a release note that describes everything at a high level?

Reply:
Thanks for reviewing my changes. I will address your other comments, but we have to agree on configuration strategy first. Success rate algorithm uses 3 parameters: success_rate_minimum_hosts, success_rate_request_volume and success_rate_stdev_factor. The first 2 are self-explanatory. The stdev is used to calculate the success threshold. If a node's success rate is below threshold it will be candidate for ejection. The final decision about ejecting is determined by enforcing_success_rate parameter: 0 means - never eject and 100 means always eject. Value in between will be used by random number generator to provide "probability" of ejection. 
In the code there are now 2 instances of SuccessRate classes. One tracks failure rate of external events and the other tracks failure rate of local origin events. To avoid config pollution, the mentioned 3 parameters: success_rate_minimum_hosts, success_rate_request_volume and success_rate_stdev_factor are used to initialize both instances of SuccessRate classes. There is however enforcing parameter for each type of SuccessRate (external and local origin). Sure, I could repeat those 3 parameters and changed their name to local_origin_xxx, but will anyone ever use it?
With the current parameters one can configure:
- turn off one type of Success Rate (set enforcing of one type to zero)
- turn on both (set enforcing of both types to values greater than zero)
- make probability of ejection for one type of SuccessRate to be more likely to happen

The only scenario which cannot be configured is to have a different minimum number hosts, volume or ejection threshold for each SuccessRate type. I doubt anyone would try to configure such scenario, but I can duplicate those config items with local_origin_ prefix if you think it makes sense.

> I'm confused as to whether this PR is going to have user visible effect for people that make no config changes.

I think that when a user uses just default values there will be no visible effect. I wrote "I think" because previously failures went to one bucket and now go into two buckets (external and local origin) and if volume of requests is small, one can imagine corner case that the number of failures in each bucket will not be sufficient to eject a host. I have to experiment with this more. 

If a  user modified enforcing parameter, that setting will now apply only for external origin SuccessRate. For internal origin SuccessRate default of 100 will be used.

Reply:
OK thanks for the explanation. The first part about how you only have an enforcing setting makes sense to me. My strong preference here would be to figure out a way of making it so that there is no impact to existing users if they don't change any config.

I'm wondering if we should have this new behavior be opt-in? For example, there could be a parameter along the lines of `split_external_local_origin_errors` or something like that. If this is not true (default) there is no change in behavior, otherwise we split and then have different settings? I actually think most users are not going to want this split at all.

Reply:
OK. I can add this param to provide no-change-in-behavior for existing users. As for the name it could be `split_external_local_origin_error`or maybe `distinguish_local_origin_errors`. I will use your suggestion. Default would be `false`. 
I will try implement it such way that depending on the value of that parameter call chain will be set up differently, so that flag does not have to be checked each time a packet is processed.
Not sure when I will finish it, as I have to travel within new few days, but this is my main focus now.

Reply:
Thanks, I think this will provide the best configuration experience for most users.

Reply:
Agree. Sorry that I cannot finish it very soon, but as I said before I will be travelling.